core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts

```diff
@@ -5,38 +5,46 @@ import { Injectable } from '@angular/core';
 
 import { switchMap, tap, withLatestFrom } from 'rxjs/operators';
 
-import { ComponentStatus, DotCMSContentlet } from '@dotcms/dotcms-models';
+import { ComponentStatus } from '@dotcms/dotcms-models';
 
 import { PromptType } from './ai-image-prompt.models';
 
-import { DotAiService } from '../../shared';
+import { AIImageSize, DotAiService } from '../../shared';
+import { AIImagePrompt, DotGeneratedAIImage } from '../../shared/services/dot-ai/dot-ai.models';
 
 const DEFAULT_INPUT_PROMPT: PromptType = 'input';
 
 export interface DotAiImagePromptComponentState {
     showDialog: boolean;
-    selectedPromptType: PromptType | null;
     prompt: string | null; // we always have the final prompt here
     editorContent: string | null;
-    contentlets: DotCMSContentlet[] | [];
+    images: DotGeneratedAIImage[];
     status: ComponentStatus;
     error: string;
+    selectedImage: DotGeneratedAIImage | null;
+    galleryActiveIndex: number;
+    orientation: AIImageSize;
 }
 
 export interface VmAiImagePrompt {
-    selectedPromptType: PromptType | null;
     showDialog: boolean;
     status: ComponentStatus;
+    images: DotGeneratedAIImage[];
+    galleryActiveIndex: number;
+    orientation: AIImageSize;
+    isLoading: boolean;
 }
 
 const initialState: DotAiImagePromptComponentState = {
-    selectedPromptType: null,
     showDialog: false,
     status: ComponentStatus.INIT,
-    contentlets: [],
+    images: [],
     prompt: null,
     editorContent: null,
-    error: ''
+    error: '',
+    selectedImage: null,
+    galleryActiveIndex: 0,
+    orientation: '1792x1024'
```

Comment:
Could be nice to use here an enum for the sizes.

Reply:
Why `orientation`? With `orientation` I would expect something like `landscape` or `portrait` not pixels size

Reply:
moved to enum to be more descriptive